web/api/queries/query.c

```diff
@@ -1581,7 +1582,11 @@ RRDR *rrd2rrdr(
     rrd_update_every = st->update_every;
     absolute_period_requested = rrdr_convert_before_after_to_absolute(&after_requested, &before_requested,
                                                                       rrd_update_every, first_entry_t,
-                                                                      last_entry_t);
+                                                                      last_entry_t, options);
+    if (options & RRDR_OPTION_ALLOW_PAST)
+        if (first_entry_t > after_requested)
+            first_entry_t = after_requested;
+
```

Comment:
This code should probably move after `rrdr_convert_before_after_to_absolute()` has been called, because only then is `after_requested` up-to-date for the comparison to make sense. `rrdr_convert_before_after_to_absolute()` can even swap `before_requested` and `after_requested` values.

Reply:
It is called after rrdr_convert_before_after_to_absolute() in L1583. 

Now there are other calls to rrdr_convert_before_after_to_absolute() when dbengine is enabled, but I don't see those affecting the `after_requested` and `before_requested` once those are set to absolute values.

Reply:
Ah yes, funny how I missed that.